runtime/src/main/scala/com/soundcloud/twinagle/ServerBuilder.scala

```diff
@@ -26,38 +27,46 @@ class ServerBuilder private (
     */
   def register[T: AsProtoService](svc: T): ServerBuilder = {
     val protoService = implicitly[AsProtoService[T]].asProtoService(svc)
-    new ServerBuilder(extension, endpoints ++ protoService.rpcs, prefix)
+    new ServerBuilder(extension, messageFilters, endpoints ++ protoService.rpcs, prefix)
   }
 
   /** withPrefix configures the HTTP path prefix to use for this server (default: `/twirp`).
     * Paths must be absolute and may not end with `/`.
     * Use an empty string to expose endpoints at the root of the HTTP path.
     */
   def withPrefix(prefix: String): ServerBuilder = {
-    new ServerBuilder(extension, endpoints, prefix)
+    new ServerBuilder(extension, messageFilters, endpoints, prefix)
+  }
+
+  def withMessageFilters(filters: Seq[MessageFilter]): ServerBuilder = {
```

Comment:
maybe just take a single `MessageFilter` here? If it becomes necessary to supply multiple filters, it'd be pretty easy to add a `andThen` method to `MessageFilter` that allows combining them, similar to the one in finagle's `Filter`